### PR TITLE
remove discord-rpc in favor of discord gamesdk

### PR DESCRIPTION
**Pull request by IllusionMan1212**
reasons for removing discord-rpc
- discord-rpc was deprecated by discord
- it didn't compile on debug

gamesdk code is proprietary so we can't add it as submodule neither can we build the binaries ourselves, so this is the best we can do now

### CODE REVIEW

File: vita3k/CMakeLists.txt [hunk 1]
```diff
@@ -123,7 +127,8 @@ if(APPLE)
 		POST_BUILD
 		COMMAND ${CMAKE_COMMAND} -E copy_directory "${CMAKE_CURRENT_SOURCE_DIR}/../data" "$<TARGET_FILE_DIR:vita3k>/../Resources/data"
 		COMMAND ${CMAKE_COMMAND} -E copy_directory "${CMAKE_CURRENT_SOURCE_DIR}/shaders-builtin/" "$<TARGET_FILE_DIR:vita3k>/../Resources/shaders-builtin"
-		COMMAND ${CMAKE_COMMAND} -E copy_directory "${PROJECT_SOURCE_DIR}/external/sdl/macos/SDL2.framework" "$<TARGET_FILE_DIR:vita3k>/../Frameworks/SDL2.framework")
+		COMMAND ${CMAKE_COMMAND} -E copy_directory "${PROJECT_SOURCE_DIR}/external/sdl/macos/SDL2.framework" "$<TARGET_FILE_DIR:vita3k>/../Frameworks/SDL2.framework"
+		COMMAND ${CMAKE_COMMAND} -E copy_if_different "${PROJECT_SOURCE_DIR}/external/discord/lib/discord_game_sdk.bundle" "$<TARGET_FILE_DIR:vita3k>/../Resources")
```
Comment (IllusionMan1212) [inline vita3k/CMakeLists.txt:131]: i'm not sure if this is the correct directory to put the binary in ?

**Comment by Zangetsu38** [comment]
Yes but is strange; discord rpc continue works for visual stidio 🤔

**Comment by sunho** [comment]
It looks like there's some headache in including sdk code in gpl project, even though sdk license is totally free. (we're distributing also their code as gpl license) Can you use cmake script to download it from the official web site? 

Some example: https://github.com/Vita3K/Vita3K/pull/837/files#diff-af3b638bc2a3e6c650974192a53c7291R58

**Comment by IllusionMan1212** [comment]
gamesdk files removed :+1:

**Comment by sunho** [comment]
You can use CMAKE_BINARY_DIR to avoid it in gitignore. Just replace CMAKE_SOURCE_DIR with it.

**Comment by sunho** [comment]
Otherwise lgtm

**Comment by sunho** [comment]
👍

**Review by sunho — APPROVED** [review]

File: external/CMakeLists.txt [hunk 2]
```diff
@@ -139,18 +139,38 @@ target_link_libraries(yaml INTERFACE yaml-cpp)
 add_library(rpcs3 INTERFACE)
 target_include_directories(rpcs3 INTERFACE "${CMAKE_CURRENT_SOURCE_DIR}/rpcs3/include")
 
-add_library(discord-rpc INTERFACE)
-if (USE_DISCORD_RICH_PRESENCE)
-	target_compile_definitions(discord-rpc INTERFACE DISCORD_RPC=NDEBUG)
-endif()
-target_include_directories(discord-rpc INTERFACE "${CMAKE_CURRENT_SOURCE_DIR}/discord-rpc/include")
+message(STATUS "Downloading discord gamesdk...")
+file(DOWNLOAD https://dl-game-sdk.discordapp.net/2.5.6/discord_game_sdk.zip
```
Comment (ichisadashioko) [inline external/CMakeLists.txt:143]: We probably can build `boost` with `CI`, publish it on the release page and add the download URL here.

**Pull request merged**